kv/kvclient/kvtenant: TestScanRangeDescriptors failed

kv/kvclient/kvtenant.TestScanRangeDescriptors [failed](https://tanzanite.cluster.engflow.com/invocations/default/926764d2-b012-4547-9a9f-705b4aeb31bf?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZjbGllbnQva3Z0ZW5hbnQ6a3Z0ZW5hbnRfdGVzdA==) on release-24.1.2-rc @ [35c3e0c133c56c91399e730f49bc488cf6dd4c11](https://github.com/cockroachdb/cockroach/commits/35c3e0c133c56c91399e730f49bc488cf6dd4c11):


```
=== RUN   TestScanRangeDescriptors
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestScanRangeDescriptors3074104974
    test_log_scope.go:81: use -show-logs to present logs inline
    tenant_scan_range_descriptors_test.go:80: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvtenant_test/pkg/kv/kvclient/kvtenant/tenant_scan_range_descriptors_test.go:80
        	Error:      	"[r69:/Tenant/2{-a} [(n1,s1):1, next=2, gen=4, sticky=1718626166.934881645,0] r77:/Tenant/2{a-b} [(n1,s1):1, next=2, gen=5, sticky=9223372036.854775807,2147483647] r80:/Tenant/2{b-/Table/11} [(n1,s1):1, next=2, gen=5, sticky=9223372036.854775807,2147483647] r71:/Tenant/2/Table/1{1-2} [(n1,s1):1, next=2, gen=4] r72:/Tenant/2/Table/{12-25} [(n1,s1):1, next=2, gen=5] r73:/Tenant/2/Table/2{5-6} [(n1,s1):1, next=2, gen=6] r74:/Tenant/2/Table/2{6-7} [(n1,s1):1, next=2, gen=7] r75:/Tenant/2/Table/2{7-8} [(n1,s1):1, next=2, gen=8] r76:/Tenant/2/Table/{28-45} [(n1,s1):1, next=2, gen=9] r78:/Tenant/2/Table/4{5-6} [(n1,s1):1, next=2, gen=10] r79:/Tenant/2/Table/4{6-7} [(n1,s1):1, next=2, gen=11] r81:/Tenant/2/Table/4{7-8} [(n1,s1):1, next=2, gen=12] r82:/Tenant/{2/Table/48-3} [(n1,s1):1, next=2, gen=12]]" should have 3 item(s), but has 13
        	Test:       	TestScanRangeDescriptors
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestScanRangeDescriptors3074104974
--- FAIL: TestScanRangeDescriptors (14.21s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestScanRangeDescriptors.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-39617

--- a/pkg/kv/kvclient/kvtenant/tenant_scan_range_descriptors_test.go
+++ b/pkg/kv/kvclient/kvtenant/tenant_scan_range_descriptors_test.go
@@ -30,13 +30,9 @@ func setup(
 	t *testing.T, ctx context.Context,
 ) (*testcluster.TestCluster, serverutils.ApplicationLayerInterface, rangedesc.IteratorFactory) {
 	tc := testcluster.StartTestCluster(t, 1, base.TestClusterArgs{
+		ReplicationMode: base.ReplicationManual,
 		ServerArgs: base.TestServerArgs{
 			DefaultTestTenant: base.TestControlsTenantsExplicitly,
-			Knobs: base.TestingKnobs{
-				Store: &kvserver.StoreTestingKnobs{
-					DisableMergeQueue: true,
-				},
-			},
 		},
 	})
 

